df.announcements.xml

```diff
@@ -463,7 +463,7 @@
         <int32_t name='unk_v40_2' init-value='-1' comment='same as unknown field in report'/>
         <int32_t name='speaker_id' ref-target='unit'/>
         <bitfield base-type='uint8_t' name='flags'>
-            <flag-bit name='sparring'/>
+            <flag-bit name='not_sparring'/>
```

Comment:
This feels to me like we're missing something - I'd have a hard time explaining why DF would implement it like this.

Is it possible that this is actually a 2-bit field? e.g. 00 = combat, 01 = sparring, 10 = hunting

Also, how did you determine this? Disassembly? There don't appear to be any references to it in df-structures (since it's stack-allocated only in DF, per the struct-type's comment)

Reply:
Yes, disassembly:
```cmp     cs:_gamemode, 0
jnz     loc_140075289                   ; if gamemode != DWARF then {check adventure mode} else
movsx   rax, [rbx+report_init.type]     ; announcement_type into rax
lea     rcx, _d_init
test    byte ptr [rcx+rax*4+d_init.announcements.flags.bitfield], 20h
jz      loc_140074FC4                   ; if d_init.announcements.flags[announcement_type].UNIT_COMBAT_REPORT == false then {check UNIT_COMBAT_REPORT_ALL_ACTIVE} else
mov     rcx, [rbx+report_init.unit1]
test    rcx, rcx
jz      loc_140074EE7                   ; if unit1 == nil then {check unit2} else
test    [rbx+report_init.flags.bitfield], 1
jz      short loc_140074E59             ; if report_init.flags.not_sparring == true then {check hunting job} else
lea     rdx, [rdi+report.id]
add     rcx, unitst.reports.log.ptr+18h
call    sub_14000BC60                   ; unit1.reports.log.Sparring:push(report.id)
mov     rcx, [rbx+report_init.unit1]
mov     eax, cs:_cur_year
mov     [rcx+(unitst.reports.last_year+4)], eax ; unit1.reports.last_year.Sparring = cur_year
mov     rcx, [rbx+report_init.unit1]
mov     eax, cs:_cur_year_tick
mov     [rcx+(unitst.reports.last_year_tick+4)], eax ; unit1.reports.last_year_tick.Sparring = cur_year_tick
or      cs:_world.status.flags.bitfield, 4 ; world.status.flags.sparring = true
jmp     loc_140074EE4                   ; {success = true; check unit2}
; ---------------------------------------------------------------------------

loc_140074E59:                          ; CODE XREF: announce_140074660+7BB↑j
mov     rdx, [rcx+unitst.job.current_job] ; --check hunting job
test    rdx, rdx
jz      short loc_140074EAD             ; if unit1.job.current_job == NULL then {push Combat report} else
mov     eax, r13d                       ; rax = false
cmp     [rdx+job.job_type], 18h
setz    al                              ; if job_type == Hunt then rax = true
test    eax, eax
jz      short loc_140074EAD             ; if rax == false then {push Combat report} else
lea     rdx, [rdi+report.id]
add     rcx, unitst.reports.log.ptr+30h
call    sub_14000BC60                   ; unit1.reports.log.Hunting:push(report.id)
```
Hunting vs Combat is determined solely based on each unit's job. (The prey considers it Combat unless they've also got the Hunt job somehow.)

Reply:
The bit could also be called "lethal_combat" to avoid confusion from the double negative when false. That might be mixed up with a combat escalation level, though.

Reply:
I like "lethal_combat". Or "enemy_combat", "hostile_combat", or something along those lines. The confusion that could result from a double-negative leads me to suspect Toady has a name for it that isn't a negative.

Reply:
"hostile_combat" sounds good.